tests/foreman/api/test_repository.py

```diff
@@ -1425,6 +1425,32 @@ def test_positive_bulk_cancel_sync(self, target_sat, module_manifest_org):
             )
             assert result.status == 0
 
+    @pytest.mark.tier2
+    @pytest.mark.parametrize(
+        'repo_options',
+        **parametrized([{'content_type': 'yum', 'url': repo_constants.CUSTOM_RPM_SHA}]),
+        indirect=True,
+    )
+    def test_positive_sync_sha_repo(self, repo, target_sat):
+        """Sync a 'sha' repo successfully
+
+        :id: b842a21d-639a-48aa-baf3-9244d8bc1415
+
+        :parametrized: yes
+
+        :customerscenario: true
+
+        :BZ: 2024889
+
+        :SubComponent: Candlepin
+        """
+        sync_result = repo.sync()
+        assert sync_result['result'] == 'success'
+        result = target_sat.execute(
+            'grep "Artifact() got an unexpected keyword argument" /var/log/messages'
+        )
+        assert result.stdout == ''
```

Comment:
asserting against the response code is going to be stronger here, because stdout could be empty, stderr have content, and grep returned a 2.

You'd want to assert the response code is 1 in this case - that grep failed to find matching lines.

Reply:
@mshriver Do you mean assert against `result.status == 1`?

Reply:
correct, asserting that grep didn't find any matches (not only that it didn't output to stdout)